ocaml/testsuite/tests/formatting/test_locations.dlocations.ocamlc.reference

```diff
@@ -88,14 +88,14 @@ let rec fib = function | 0 | 1 -> 1 | n -> (fib (n - 1)) + (fib (n - 2))
       <def>
         pattern (test_locations.ml[17,534+8]..test_locations.ml[17,534+11])
           Tpat_var "fib"
-          value_mode global,many,shared
+          value_mode meet_local,once(0[global,many,global,many]),join_shared(1[shared,shared])
```

Comment:
The changes in this file seem worrying to me. That is, the change proposed here means variables will live for longer than they did previously, but I'm not sure of the upside. I understand we want to be able to set the legacy modes separately, but it's not obvious to me that the step in this PR is necessary.

Reply:
This is caused by `check_const` now requiring the mode to be allowed on both LHS and RHS, because it calls `zap_to_floor_morphvar` which requires that. the mode on patterns is left-only so cannot be `check_const`, and therefore we are printing them raw.

I think this doesn't affect anything on the production path - `check_const` is only called for tests and debug printing.

Reply:
Not quite: `check_const` is also used to extract the mode from arrow types in `external` declarations. But I agree that we needn't worry about that case. OK -- I'm much happier with this patch now.